src/Util/Sniffs/Orthography.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Utility functions for checking the orthography of arbitrary text strings.
+ *
+ * @author    Juliette Reinders Folmer <phpcs_nospam@adviesenzo.nl>
+ * @copyright 2019 Juliette Reinders Folmer. All rights reserved.
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Util\Sniffs;
+
+class Orthography
```

Comment:
I fundamentally disagree. I struggled with this quite a bit when I created the code and I originally had them in the `TextString` class.

These methods, however, are not only for text string and/or comments.
The `isFirstChar*()` methods can also be used by the `ConstructNames` methods regarding various naming conventions.
I just haven't got round to writing/refactoring those `ConstructNames` methods.

So, to me, having them separate and available for use within different contexts - be it comments, text strings or construct names, makes much more sense.

Reply:
I get what you are saying, but I can't see developers going looking for these sort of methods anywhere but a string util class. I suspect most wouldn't even know what orthography means, so the naming of the class is confusing.

Does it really matter that a util class for working with strings is a collection of functions that work for various types of strings, be they function names, quoted strings, or inline HTML? Could we come up with a new name for this collection of methods that isn't `TextString`?

Reply:
I must admit I had to google `ortography` 😄 

At one hand, having one class do multiple things might be a bit too much (SRP), but after looking at the `TextString` class, it's not too big and _technically_ comments/regular strings and names of various constructs are all strings, so this could be fitted in that one class for the ease of use.

I mean, if I were to need some utility regarding text (be it punctuation or first character helper), ortography is not something that would immediately come to mind.

Reply:
While I too understand why these were broken out, I also agree that in terms of being able to see at a glance what utility methods are available, these fall under the strings category (or at least need wrapper methods in that TextStrings class that uses the logic contained here).

Reply:
I get that it could be beneficial maintenance wise by breaking down and defining the types of strings, Ortogrophy, Case, HTML, Comment etc. However I think a wrapper for quick (lazy) access when actually writing standards would be helpful. I extended Common.php to add some checks on a standard I wrote, would have been nice to have StringUtil or something like that to extend.